framework/src/variables/MooseVariableFV.C

```diff
@@ -1051,27 +1098,33 @@ MooseVariableFV<OutputType>::uncorrectedAdGradSln(const FaceInfo & fi) const
   // gradient
   if (elem_two && this->hasBlocks(elem_two->subdomain_id()))
   {
-    const VectorValue<ADReal> & elem_two_grad = adGradSln(elem_two);
+    const VectorValue<ADReal> & elem_two_grad = adGradSln(elem_two, correct_skewness);
+
+    const auto interp_method =
+        (_face_interp_method == Moose::FV::InterpMethod::Average ||
+         _face_interp_method == Moose::FV::InterpMethod::SkewCorrectedAverage)
+            ? _face_interp_method
+            : Moose::FV::InterpMethod::Average;
```

Comment:
Yes, if vertex-based interpolation is specified, we use the average to interpolate the gradients. This is only because the different gc_-s. The skew-corrected scenario needs the skew-corrected gcs for this interpolation as well. Also, I think using vertex-based interpolation for the gradients would be super expensive.